src/test/java/io/castle/client/CastleNoTrackOptionTest.java

```diff
@@ -0,0 +1,69 @@
+package io.castle.client;
+
+import io.castle.client.api.CastleApi;
+import io.castle.client.model.AuthenticateAction;
+import io.castle.client.model.AuthenticateFailoverStrategy;
+import io.castle.client.model.Verdict;
+import okhttp3.mockwebserver.MockResponse;
+import org.assertj.core.api.Assertions;
+import org.junit.Assert;
+import org.junit.Test;
+import org.springframework.mock.web.MockHttpServletRequest;
+
+import javax.servlet.http.HttpServletRequest;
+
+public class CastleNoTrackOptionTest extends AbstractCastleHttpLayerTest {
+
+    public CastleNoTrackOptionTest() {
+        super(new AuthenticateFailoverStrategy(AuthenticateAction.CHALLENGE));
+    }
+
+    @Test
+    public void noTrackOptionDisableRequests() throws InterruptedException {
+
+        //Given a mock HTTP request is provided
+        HttpServletRequest request = new MockHttpServletRequest();
+        //And a CastleApi is created with doNotTrack option
+        CastleApi castleApi = sdk.onRequest(request).doNotTrack(true);
+
+        //When all API call are executed
+        castleApi.track("testEvent");
+        castleApi.identify("userId", true);
+        Verdict verdict = castleApi.authenticate("testEvent", "userId");
+        //and we await the timeout period to avoid concurrency races
+        Thread.sleep(120);
+
+        //Then no calls are made to the backend
+        Assertions.assertThat(server.getRequestCount()).isEqualTo(0);
+        //And the Verdict is the default ALLOW value
+        Assert.assertEquals(AuthenticateAction.ALLOW, verdict.getAction());
+        Assert.assertEquals("userId", verdict.getUserId());
+    }
+
+    @Test
+    public void trackOptionAllowRequests() throws InterruptedException {
+
+        //Given a mock HTTP request is provided
+        HttpServletRequest request = new MockHttpServletRequest();
+        //And responses are setup on mock server
+        server.enqueue(new MockResponse().setResponseCode(200));
+        server.enqueue(new MockResponse().setResponseCode(200));
+        server.enqueue(new MockResponse().setBody(
+                "{\n" +
+                        "  \"action\": \"deny\",\n" +
+                        "  \"user_id\": \"12345\"\n" +
+                        "}"));
+        //And a CastleApi is created with doNotTrack option
+        CastleApi castleApi = sdk.onRequest(request).doNotTrack(false);
+        //When all API call are executed
+        castleApi.track("testEvent");
+        server.takeRequest();
+        castleApi.identify("userId", true);
+        server.takeRequest();
+        castleApi.authenticate("testEvent", "userId");
```

Comment:
please add a test that `authenticate`  will return an `APPROVE` verdict. And failover reason `Castle set to do not track.` See this PR in ruby https://github.com/castle/castle-ruby/pull/55/files

Reply:
ok. Added.